packages/pharos/src/components/toggle-button-group/pharos-toggle-button-group.ts

```diff
@@ -0,0 +1,164 @@
+import { html, LitElement } from 'lit-element';
+import type { TemplateResult, CSSResultArray } from 'lit-element';
+import { toggleButtonGroupStyles } from './pharos-toggle-button-group.css';
+import { designTokens } from '../../styles/variables.css';
+import { customElement } from '../../utils/decorators';
+import type { PharosToggleButton } from './pharos-toggle-button';
+
+const matchesFunc = 'matches' in Element.prototype ? 'matches' : 'msMatchesSelector';
+
+/**
+ * Pharos toggle button group component.
+ *
+ * @element pharos-toggle-button-group
+ *
+ * @slot - Contains the buttons.
+ *
+ */
+@customElement('pharos-toggle-button-group')
+export class PharosToggleButtonGroup extends LitElement {
+  public static get styles(): CSSResultArray {
+    return [designTokens, toggleButtonGroupStyles];
+  }
+
+  protected firstUpdated(): void {
+    this.addEventListener('pharos-toggle-button-selected', this._handleButtonSelected);
+    this.addEventListener('keydown', this._handleKeydown);
+    this.addEventListener('focusout', this._handleFocusout);
+
+    const toggleButtons = this.querySelectorAll(
+      `pharos-toggle-button`
+    ) as NodeListOf<PharosToggleButton>;
+    this._selectInitialToggleButton(toggleButtons);
+
+    const maxIdx = toggleButtons.length - 1;
+    toggleButtons.forEach((button, idx) => {
+      button['_first'] = idx == 0;
+      button['_last'] = idx == maxIdx;
+    });
+  }
+
+  private _selectInitialToggleButton(toggleButtons: NodeListOf<PharosToggleButton>): void {
+    const selected = this.querySelector(`pharos-toggle-button[selected]`) as PharosToggleButton;
+    const selectedButton: PharosToggleButton = selected ? selected : toggleButtons[0];
+
+    selectedButton.selected = true;
+  }
+
+  private _handleButtonSelected(event: Event): void {
+    const selected = event.target as PharosToggleButton;
+
+    const previous = this.querySelector(
+      `pharos-toggle-button[selected]:not([id="${selected.id}"])`
+    ) as PharosToggleButton;
+
+    if (previous) {
+      previous.selected = false;
+    }
+
+    const toggleButtons = Array.prototype.slice.call(
+      this.querySelectorAll(`pharos-toggle-button`)
+    ) as PharosToggleButton[];
+    let selectIdx = -1;
```

Comment:
Not sure if it would be useful, but might be able to use `.findIndex()` here.

Reply:
fixed